vendor/ansible/roles/swarm/templates/swarm.j2

```diff
@@ -8,12 +8,26 @@ fi
 
 case $1 in
 start)
-    echo starting swarm as on {{ node_name }}[{{ node_addr }}]
-    {% if swarm_bootstrap_node_name == node_name %}
-        /usr/bin/swarm manage -H tcp://{{ node_addr }}:{{ swarm_api_port }} etcd://{{ node_addr }}:{{ etcd_client_port1 }} &
+    echo starting swarm as {{ run_as }} on {{ node_name }}[{{ node_addr }}]
+    {% if run_as == "master" -%}
+    /usr/bin/docker run -t -d -p {{ swarm_api_port }}:{{ swarm_api_port }} --name=swarm-manager \
+        swarm:{{ swarm_version }} manage \
+        -H :{{ swarm_api_port }} \
+        --replication --advertise={{ node_addr }}:{{ swarm_api_port }} \
+        etcd://{{ node_addr }}:{{ etcd_client_port1 }}
     {% endif %}
+    /usr/bin/docker run -t --name=swarm-agent \
```

Comment:
@mapuri so, for the master nodes, we run both `swarm manage` and `swarm join`.  we might "conditionally" want to do `swarm join` only on the `service-worker` group, in a scenario where we don't want containers on the master nodes.